packages/tx/src/baseTransaction.ts

```diff
@@ -298,6 +299,54 @@ export abstract class BaseTransaction<TransactionObject> {
     return tx
   }
 
+  /**
+   * Signs a transaction.
+   *
+   * Note that the signed tx is returned as a new object,
+   * use as follows:
+   * ```javascript
+   * const signer = (buffer: Buffer): Promise<{ signature, recid }>
+   * const signedTx = await tx.sign(signer)
+   * ```
+   */
+  signWithSigner(
+    signer: (msgHash: Buffer) => Promise<{ signature: Uint8Array; recid: number }>
+  ): Promise<TransactionObject> {
+    // Hack for the constellation that we have got a legacy tx after spuriousDragon with a non-EIP155 conforming signature
+    // and want to recreate a signature (where EIP155 should be applied)
+    // Leaving this hack lets the legacy.spec.ts -> sign(), verifySignature() test fail
+    // 2021-06-23
+    let hackApplied = false
+    if (
+      this.type === 0 &&
+      this.common.gteHardfork('spuriousDragon') &&
+      !this.supports(Capability.EIP155ReplayProtection)
+    ) {
+      this.activeCapabilities.push(Capability.EIP155ReplayProtection)
+      hackApplied = true
+    }
+
+    const msgHash = this.getMessageToSign(true)
+
+    const _processSignature = ({ signature, recid }: { signature: any; recid: number }) => {
+      const { r, s, v } = splitSignature(signature, recid)
```

Comment:
Shouldn't we pass a `chainId` here? Otherwise we will get an incorrect ` v` value. (required for Polygon & maybe EIP1559, I don't quite remember)

Reply:
The way I see it chainId is never supplied to `ecsign` here: https://github.com/ExodusMovement/ethereumjs-monorepo/blob/3b99915b702174d1847a206b615f75a6db76d6a8/packages/tx/src/baseTransaction.ts#L288

What am I missing?

Reply:
No, you're right, it's fine. `this._processSignature` will take care of it.